commit bb9a7de92c3a7a4ae8a47536e36559673dd886e9
Author: Ben Cheng <bccheng@google.com>
Date:   Tue Feb 4 14:34:21 2014 -0800

    Fix aarch64 to use Android specs.
    
    Flags include:
    - CC1_SPEC
    - CC1PLUS_SPEC
    - LIB_SPEC
    - STARTFILE_SPEC
    - ENDFILE_SPEC
    
    Change-Id: Ia96ca57ade849c964c91f5390e6c9cded16a9bf4

diff --git a/gcc/config/aarch64/aarch64-linux.h b/gcc/config/aarch64/aarch64-linux.h
index 2bdae7a..8cfe4d4 100644
--- a/gcc/config/aarch64/aarch64-linux.h
+++ b/gcc/config/aarch64/aarch64-linux.h
@@ -44,4 +44,25 @@
     }						\
   while (0)
 
+#undef  CC1_SPEC
+#define CC1_SPEC							\
+  LINUX_OR_ANDROID_CC (GNU_USER_TARGET_CC1_SPEC,			\
+		      GNU_USER_TARGET_CC1_SPEC " " ANDROID_CC1_SPEC("-fpic"))
+
+#define CC1PLUS_SPEC \
+  LINUX_OR_ANDROID_CC ("", ANDROID_CC1PLUS_SPEC)
+
+#undef  LIB_SPEC
+#define LIB_SPEC							\
+  LINUX_OR_ANDROID_LD (GNU_USER_TARGET_LIB_SPEC,			\
+		       GNU_USER_TARGET_NO_PTHREADS_LIB_SPEC " " ANDROID_LIB_SPEC)
+
+#undef	STARTFILE_SPEC
+#define STARTFILE_SPEC \
+  LINUX_OR_ANDROID_LD (GNU_USER_TARGET_STARTFILE_SPEC, ANDROID_STARTFILE_SPEC)
+
+#undef	ENDFILE_SPEC
+#define ENDFILE_SPEC \
+  LINUX_OR_ANDROID_LD (GNU_USER_TARGET_ENDFILE_SPEC, ANDROID_ENDFILE_SPEC)
+
 #endif  /* GCC_AARCH64_LINUX_H */
--- gcc-4.9/gcc/config.gcc.omv~	2014-03-06 07:54:19.389906324 +0100
+++ gcc-4.9/gcc/config.gcc	2014-03-06 07:54:26.963915051 +0100
@@ -903,7 +903,8 @@ aarch64*-*-elf)
 	TM_MULTILIB_CONFIG=`echo $TM_MULTILIB_CONFIG | sed 's/^,//'`
 	;;
 aarch64*-*-linux*)
-	tm_file="${tm_file} dbxelf.h elfos.h gnu-user.h linux.h glibc-stdint.h"
+	tm_file="${tm_file} dbxelf.h elfos.h gnu-user.h linux.h linux-android.h glibc-stdint.h"
 	tm_file="${tm_file} aarch64/aarch64-elf.h aarch64/aarch64-linux.h"
 	tmake_file="${tmake_file} aarch64/t-aarch64 aarch64/t-aarch64-linux"
+	extra_options="${extra_options} linux-android.opt"
 	case $target in
diff --git gcc-4.8/gcc/config/aarch64/aarch64-linux.h gcc-4.8/gcc/config/aarch64/aarch64-linux.h
index 83efad4..2bdae7a 100644
--- gcc-4.8/gcc/config/aarch64/aarch64-linux.h
+++ gcc-4.8/gcc/config/aarch64/aarch64-linux.h
@@ -40,6 +40,7 @@
   do						\
     {						\
 	GNU_USER_TARGET_OS_CPP_BUILTINS();	\
+	ANDROID_TARGET_OS_CPP_BUILTINS();	\
     }						\
   while (0)
 
